include/arch/arm/cortex_m/mpu/arm_mpu.h

```diff
@@ -102,18 +102,18 @@ struct arm_mpu {
 
 /* Some helper defines for common regions */
 #define REGION_USER_RAM_ATTR(size) \
-		(NORMAL_OUTER_INNER_NON_CACHEABLE_NON_SHAREABLE | \
+		(NORMAL_OUTER_INNER_WRITE_BACK_NON_SHAREABLE | \
```

Comment:
Since caches are being explicitly enabled during SoC startup, would this cause issues if the cache isn't enabled in a particular SoC that doesn't have that for some reason?

Reply:
No.  But it probably shouldn't be enabled by default for all arches.  This definitely feels like a SoC specific thing as most ARM SoCs don't have I/D cache.

Reply:
Can we create new macros with cache enabled? which we can add for SAM e70.
So that the functionality remains then same on M3/M4 SOCs which are using this at the moment.

Reply:
Also, keep in mind that: these macros, here, are **only** used when configuring the whole RAM, FLASH during MPU initialization. Regions for user-space, app-memory, stack-guards are configured in run-time, and, currently, they enforce non-cacheable configuration. So this change, here, is not doing the whole thing, IMHO.

Reply:
@ioannisg, for what I understand from the code, the  user-space, app-memory and stack-guards are configured in `arch/arm/core/cortex_m/mpu/arm_mpu.c` function `_get_region_attr_by_type`. The cache is the fourth argument of the `_get_region_attr` function, and is set to 1. So I believe everything fine on that aspect.

About `REGION_USER_RAM_ATTR`, I agree it's not used, it's probably better to just remove it.

Reply:
@aurel32 right, this gives write-through, no-write-allocate, non-sharable attributes. Since those are RAM regions, I wonder if we could also switch to Write-Back policy.

Also, REGION_USER_RAM_ATTR is currently not used, but probably good to keep it

Reply:
@ioannisg, you're correct, switching to write-back indeed makes sense, maybe except for the stack-guard, but that won't hurt.

About REGION_USER_RAM_ATTR, what's the use case? Isn't it redundant with CONFIG_APPLICATION_MEMORY N